app/models/app_remote_kill.rb

```diff
@@ -0,0 +1,65 @@
+class AppRemoteKill < ActiveRecord::Base
+  belongs_to :app
+  belongs_to :split
+
+  attribute :first_bad_version, :app_version
+  attribute :fixed_version, :app_version
+
+  validates :app, :split, :reason, :override_to, :first_bad_version, presence: true
+  validates :reason, uniqueness: { scope: %i(app split) }
+
+  validate :override_to_must_exist
+  validate :fixed_version_must_be_greater_than_first_bad_version
+  validate :must_not_overlap_existing
+
+  scope :affecting, ->(app_build) do
+    where(
+      arel_table[:app_id].eq(app_build.app_id)
+      .and(arel_table[:first_bad_version].lteq(app_build.version))
+      .and(arel_fixed_version_is_null_or_greater_than(app_build.version))
+    )
+  end
+
+  scope :overlapping, ->(other) do
+    where(app_id: other.app_id, split_id: other.split_id)
+      .where.not(id: other.id)
+      .where(
+        if other.fixed_version.nil?
+          Arel::Nodes::True.new
```

Comment:
This is mostly to make the chaining easier - really I just want to only do the second half of this where clause if other doesn't have a fixed version. I don't think I'm costing PG much to process `TRUE AND [...]` vs `[...]`